src/v2/Apps/Artwork/Components/ArtworkSidebar/ArtworkSidebarAuctionProgressBar.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from "react"
+import { createFragmentContainer, graphql } from "react-relay"
+import { useTimer } from "v2/Utils/Hooks/useTimer"
+import { ArtworkSidebarAuctionProgressBar_artwork } from "v2/__generated__/ArtworkSidebarAuctionProgressBar_artwork.graphql"
+import { ProgressBar } from "@artsy/palette"
+
+interface ArtworkSidebarAuctionProgressBarProps {
+  artwork: ArtworkSidebarAuctionProgressBar_artwork
+}
+
+const ArtworkSidebarAuctionProgressBar: React.FC<ArtworkSidebarAuctionProgressBarProps> = ({
+  artwork,
+}) => {
+  const { sale, saleArtwork } = artwork
+
+  const extendedBiddingPeriodMinutes = sale?.extendedBiddingPeriodMinutes!
+  const biddingEndAt = saleArtwork?.extendedBiddingEndAt ?? saleArtwork?.endAt
+
+  const {
+    time: { days, hours, minutes, seconds },
+  } = useTimer(biddingEndAt!)
+
+  const parsedDaysUntilEnd = parseInt(days, 10)
+  const parsedHoursUntilEnd = parseInt(hours, 10)
+  const parsedMinutesUntilEnd = parseInt(minutes, 10)
+  const parsedSecondsUntilEnd = parseInt(seconds, 10)
+
+  // This assumes that the period before a lot's end time which will extend bidding
+  // and cause a progress bar to render is strictly less than 1 hour, which is
+  // reasonable. In practice, we expect the value of `extendedBiddingPeriodMinutes`
+  // to be 1 or 2.
+  const isWithinExtendedBiddingPeriod =
+    parsedDaysUntilEnd < 1 &&
+    parsedHoursUntilEnd < 1 &&
+    parsedMinutesUntilEnd < extendedBiddingPeriodMinutes
+
+  const percentComplete =
+    (parsedSecondsUntilEnd + parsedMinutesUntilEnd * 60) /
+    (extendedBiddingPeriodMinutes * 60)
```

Comment:
I think we want this to be extendedBiddingPeriodIntervalMinutes, as isn't the period minutes represent the time leading up to the original end of a sale in which we would extend bidding if a bid were to happen?

Reply:
I think we're actually both right on this one! It's a bit subtle-

  - once bidding has been extended, we basically always want to show this progress bar. as it ticks down to 0 w/ the realtime websocket update, it can jump back up to full and tick down again when a new bid comes in. As you point out, we might want to consider the interval tacked on for that.
  - before bidding has been extended, as the lot is approaching its closing, i think we do want to show the progress bar as an increased CTA. in this case, i think considering the period makes sense.

Given that it sounds like they'll be the same, it's probably fine either way (not to overthink it) and I'll just pick one. I'll also consider if it makes sense to use both values, before extension use the 'period' value, and after extension use the 'interval', etc.

Reply:
that sounds perfectly reasonable to me!